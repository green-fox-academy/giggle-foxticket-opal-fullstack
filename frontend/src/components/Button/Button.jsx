import React from 'react';
import './Button.styles.scss';

const styles = [
  'btn--primary--solid',
  'btn--warning--solid',
  'btn--danger--solid',
  'btn--success--solid',
  'btn--primary--outline',
  'btn--warning--outline',
  'btn--danger--outline',
  'btn--success--outline',
];

const sizes = ['btn--medium', 'btn--large'];

const Button = ({ children, type, onClick, buttonStyle, buttonSize }) => {
  const checkButtonStyle = styles.includes(buttonStyle)
    ? buttonStyle
    : styles[0];

  const checkButtonSize = sizes.includes(buttonSize) ? buttonSize : sizes[0];

  return (
    <button
      className={`btn ${checkButtonStyle} ${checkButtonSize}`}
      onClick={onClick}
      type={type}
    >
      {children}
    </button>
  );
};

export default Button;
