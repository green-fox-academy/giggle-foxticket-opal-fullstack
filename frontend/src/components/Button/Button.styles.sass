$--primary: #4628ff
$--warning: #ffd028
$--danger: #eb3f27
$--success: #75fa83
$--white: #fdfdfd
$--dark: #181717

.btn
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif
  font-weight: 400
  border-radius: 10px
  cursor: pointer
