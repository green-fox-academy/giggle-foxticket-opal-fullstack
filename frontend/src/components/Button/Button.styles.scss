$--primary: #4628ff;
$--warning: #ffd028;
$--danger: #eb3f27;
$--success: #75fa83;
$--white: #fdfdfd;
$--dark: #181717;

.btn {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  font-weight: 400;
  border: none;
  border-radius: 10px;
  cursor: pointer;

  &:hover {
    filter: brightness(0.8);
  }

  &.btn--primary--solid {
    background: $--primary;
    color: $--white;
  }

  &.btn--warning--solid {
    background: $--warning;
    color: $--dark;
  }

  &.btn--danger--solid {
    background: $--success;
    color: $--white;
  }

  &.btn--primary--outline {
    background: transparent;
    border: 2px solid $--primary;
    color: $--primary;
  }

  &.btn--warning--outline {
    background: transparent;
    border: 2px solid $--warning;
    color: $--warning;
  }

  &.btn--danger--outline {
    background: transparent;
    border: 2px solid $--danger;
    color: $--danger;
  }

  &.btn--success--outline {
    background: transparent;
    border: 2px solid $--success;
    color: $--success;
  }

  &.btn--medium {
    padding: 10px 20px;
    font-size: 18px;
  }

  &.btn--small {
    padding: 15px 30px;
    font-size: 20px;
  }
}
