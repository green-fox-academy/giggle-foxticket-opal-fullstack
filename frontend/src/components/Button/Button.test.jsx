import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

it('expect to render Button component', () => {
  expect(shallow(<Button />).length).toEqual(1);
});
