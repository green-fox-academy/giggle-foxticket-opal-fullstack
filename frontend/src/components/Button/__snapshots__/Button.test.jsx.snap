// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component should render Button component 1`] = `
<button
  className="btn btn--primary--solid btn--medium"
/>
`;
